modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/serializer/RecordV1Serializer.java

```diff
@@ -226,8 +226,11 @@ public RecordV1Serializer(
     }
 
     /** {@inheritDoc} */
-    @Override public WALRecord readRecord(FileInput in0, WALPointer expPtr) throws IOException, IgniteCheckedException {
-        return readWithCrc(in0, expPtr, recordIO);
+    @Override public WALRecord readRecord(ByteBufferBackedDataInput in0, WALPointer expPtr) throws IOException, IgniteCheckedException {
+        if (in0 instanceof FileInput)
+            return readWithCrc((FileInput)in0, expPtr, recordIO);
+        else
+            return readWithoutCrcCheck(in0, expPtr, recordIO);
```

Comment:
Why did you skip CRC check for non file input?

Reply:
This was in assumption that CRC check logic is not required, which is probably not true. I moved CRC checking logic to upper level so that all inheritors of ByteBufferBackedDataInput can now leverage the CRC check logic.